rosbag2_transport/test/rosbag2_transport/test_record_all.cpp

```diff
@@ -107,20 +113,27 @@ TEST_F(RecordIntegrationTestFixture, published_messages_from_multiple_services_a
   ASSERT_TRUE(client_manager_1->wait_for_service_to_be_ready());
   ASSERT_TRUE(client_manager_2->wait_for_service_to_be_ready());
 
+  // At this point, we expect that the services /test_service_1 and /test_service_2, along with
+  // the event topics /test_service_1/_service_event and /test_service_2/_service_event are
+  // available to be recorded.  However, wait_for_service_to_be_ready() only checks the services,
+  // not the event topics, so ask the recorder to make sure it has successfully subscribed to all.
+  ASSERT_TRUE(recorder->wait_for_topic_to_be_discovered("/test_service_1/_service_event"));
+  ASSERT_TRUE(recorder->wait_for_topic_to_be_discovered("/test_service_2/_service_event"));
```

Comment:
I don't mind to add wait for service event topics here. However, I am curious if we ever got a failure in this test.

Reply:
Yes, we have been sporadically seeing this test fail.  Locally, under load, I can make this test fail about 1% of the time.  The reason it fails is because we send two requests below, and each request is expected to make 2 callbacks happen on the `/test_service_<n>/_service_event` topic.  If we send the requests before the topics were matched in the recorder, then line 137 below fails because we never get the callbacks.